### ISSUE TITLE
[BUG] saveDraft Event not working

### ISSUE THREAD

**Issue by DonAmit197** [formio/formio.js#4921]
I have a form where I need Save as Draft and Submit buttons. I'm using the latest version of form.io javascript api.
Using the `eventListener ` `saveDraft`, but it's not working for me.
```
 form.on("saveDraft",function(submission){
    console.log(submission)
  })
```

Any help in this regard is highly appreciated.

**Comment by junelau** [comment]
the documentation in https://formio.github.io/formio.js/app/examples/savedraft.html 
doesn't match up with the behaviour I'm seeing using v4.14.12

despite having a `formioUser` in my localstorage, the form loads with the warning "Cannot restore draft because there is no formio instance." and no drafts are ever restored, and no drafts are ever saved.

**Comment by brendanbond** [comment]
Thanks @junelau & @DonAmit197 , we're aware of this issue and are working on a fix

**Comment by DerH4NNES** [comment]
The restoreDraft method ist called in the webform constructor.
`this.formio` will be defined/overwritten after setting the webforms url (triggered by setSrc or setUrl)
The formio object cannot passed by options, because `this.options.formio` is not used.

@brendanbond @DonAmit197

**Comment by DerH4NNES** [comment]
This MR seems to be the problem.
We dont wait for the ready state of webform anymore.
#https://github.com/formio/formio.js/pull/4446 #4446

**Comment by lane-formio** [comment]
We've done a lot of work recently around save as draft and I expect this issue has been resolved in formiojs@4.21.x so I'm going to go ahead and close. If there still seems to be a problem, let me know and I'll re-open.
